assets/js/phoenix_live_view/rendered.js

```diff
@@ -17,6 +17,66 @@ import {
   isCid,
 } from "./utils"
 
+export let modifyRoot = (html, attrs, innerHTML) => {
+  html = html.trimStart()
+  let tagStartsAt = null
+  let pos = 0
+  while(pos < html.length){
+    let maybeStart = html.indexOf("<", pos)
+    if(maybeStart === -1){ break }
+    if(maybeStart >= 0 && html.charAt(maybeStart + 1) !== "!"){
+      tagStartsAt = maybeStart
+      break
+    }
+    pos += html.indexOf("-->", pos)
+  }
+
+  let commentBefore = tagStartsAt === 0 ? null : html.slice(0, tagStartsAt).trim()
+  let contentAfter = null
+  html = html.slice(tagStartsAt).trimStart()
+  let tagNamesEndsAt
+  for(let i = 1; i < html.length; i++){
+    let char = html.charAt(i)
+    if([">", " ", "\n", "\t", "\r"].indexOf(char) >= 0 || (char === "!" && html.charAt(i + 1) === ">")){
+      tagNamesEndsAt = i
+      break
+    }
+  }
+
+  let tag = html.slice(1, tagNamesEndsAt)
+  let tagOpenEndsAt = html.indexOf(">")
+  let tagInnerHTML
+  let closingTag
+  let isVoid = html.charAt(tagOpenEndsAt - 1) === "/"
+  let tagOpenContent = isVoid ? html.slice(0, tagOpenEndsAt - 1) : html.slice(0, tagOpenEndsAt)
+  if(isVoid){
+    contentAfter = html.slice(tagOpenEndsAt + 1) || null
+  } else {
+    closingTag = `</${tag}>`
+    let tagInnerEndsAt = html.lastIndexOf(closingTag)
+    tagInnerHTML = html.slice(tagOpenEndsAt + 1, tagInnerEndsAt)
+    contentAfter = html.slice(tagInnerEndsAt + closingTag.length)
+  }
+
+  let newAttrs = []
+  Object.keys(attrs).forEach(attr => {
+    if(!tagOpenContent.includes(`${attr}="`)){ newAttrs.push([attr, attrs[attr]]) }
```

Comment:
This can also have false positives, albeit more unlikely. However, my suggestion is to not use the ID at all, let's just inject new/private attributes.